app/views/unattended/provisioning_templates/user_data/userdata_open_vm_tools.erb

```diff
@@ -0,0 +1,36 @@
+<%#
+kind: user_data
+name: UserData open-vm-tools
+model: ProvisioningTemplate
+oses:
+- CentOS
+- Fedora
+- Debian
+- Ubuntu
+-%>
+# Template for VMWare customization via open-vm-tools
+
+identity:
+  LinuxPrep:
+    domain: <%= @host.domain %>
+    hostName: <%= @host.shortname %>
+    hwClockUTC: true
+    timeZone: <%= host_param('time-zone') || 'UTC' %>
+
+globalIPSettings:
+  dnsSuffixList: [<%= @host.domain %>]
+  <%- @host.interfaces.each do |interface| -%>
+  <%- next unless interface.subnet -%>
+  dnsServerList: [<%= interface.subnet.dns_primary %>, <%= interface.subnet.dns_secondary %>]
```

Comment:
I suggest using the line below as the DNS primary and secondary is not mandatory and on its absence, it will create an invalid YAML file with the output `[ , ]`. 
```erb
dnsServerList: [<%=  [@host.subnet.dns_primary, @host.subnet.dns_secondary].reject { |c| c.empty? }.join(',') %>]
```

Reply:
In other places we also use the DNS as a list and reject empty values. I think a helper in Foreman that provides a list of non-empty values (or an empty list) would make sense.

Reply:
I created https://projects.theforeman.org/issues/26250, PR will follow tonight.

Reply:
https://github.com/theforeman/foreman/pull/6541